07-routing-gossip.md

```diff
@@ -727,6 +727,155 @@ is simple to implement.
+elsewhere for additional data.
+
+The explicit `reply_short_channel_ids_extended_end` message means that the receiver can
+indicate it doesn't know anything, and the sender doesn't need to rely on
+timeouts.  It also causes a natural ratelimiting of queries.
+
+### The `query_channel_range_extended` and `reply_channel_range_extended` Messages
+
+1. type: 1013 (`query_channel_range_extended`) (`gossip_queries`)
+2. data:
+    * [`32`:`chain_hash`]
+    * [`4`:`first_blocknum`]
+    * [`4`:`number_of_blocks`]
+
+1. type: 1014 (`reply_channel_range_extended`) (`gossip_queries`)
+2. data:
+    * [`32`:`chain_hash`]
+    * [`4`:`first_blocknum`]
+    * [`4`:`number_of_blocks`]
+    * [`1`:`complete`]
+    * [`2`:`len`]
+    * [`len`:`encoded_channel_digest_info`]
+
+This allows a query for channels within specific blocks.
+
+#### Requirements
+
+The sender of `query_channel_range_extended`:
+  - MUST NOT send this if it has sent a previous `query_channel_range_extended` to this peer and not received all `reply_channel_range_extended` replies.
+  - MUST set `chain_hash` to the 32-byte hash that uniquely identifies the chain
+  that it wants the `reply_channel_range_extended` to refer to
+  - MUST set `first_blocknum` to the first block it wants to know channels for
+  - MUST set `number_of_blocks` to 1 or greater.
+
+The receiver of `query_channel_range_extended`:
+  - if it has not sent all `reply_channel_range_extended` to a previously received `query_channel_range_extended` from this sender:
+    - MAY fail the connection.
+  - MUST respond with one or more `reply_channel_range_extended` whose combined range
+	cover the requested `first_blocknum` to `first_blocknum` plus
+	`number_of_blocks` minus one.
+  - For each `reply_channel_range_extended`:
+    - MUST set with `chain_hash` equal to that of `query_channel_range_extended`,
+    - MUST encode channel digest information, for every open channel it knows in blocks `first_blocknum` to `first_blocknum` plus `number_of_blocks` minus one.
+    - MUST limit `number_of_blocks` to the maximum number of blocks whose
+      results could fit in `encoded_channel_digest_info`
+    - if does not maintain up-to-date channel information for `chain_hash`:
+      - MUST set `complete` to 0.
+    - otherwise:
+      - SHOULD set `complete` to 1.
+
+Channel digest information for each channel is encoded as follow:
+- `short_channel_id`
+- timestamp of the `channel_update` for node 1
+- timestamp of the `channel_update` for node 2
+- checksum for the `channel_update` for node 1
+- checksum for the `channel_update` for node 2
+
+The checksum for a `channel_update` is a CRC32 checksum that covers the following fields: `short_channel_id`, `channel_flags`, `cltv_expiry_delta`, `fee_base_msat`, `fee_proportional_millionths`. It can be used to avoid querying new updates that do not include new information, and does not cover static fields such as `htlc_minimum_msat` or `htlc_maximum_msat`.
```

Comment:
The serialization for these needs to be defined, i.e., we need to have the same message format definition as for any other wire message.